### PR TITLE
fix: Improve reliability of embedded resources resolving

### LINKED ISSUE #1501: Bug? in Mono.Cecil causes confusing warnings/errors in CLI runner
**Describe the bug**
When running on a .NET framework project, I'm getting a confusing warning and the same exception is raised 3 times. After this, it properly runs the tests with the mutations.

**Logs**
[16:39:25 INF] Identifying project to mutate.
[16:39:27 INF] The project [path to class library I want mutated] will be mutated.
[16:39:29 INF] Analysis complete.
[16:39:29 INF] Building test project [Path to project with unit tests] (1/1)
[16:39:45 INF] Initializing test runners (VsTest)
[16:39:49 INF] Total number of tests found: 28
[16:39:49 INF] Initial testrun started
[16:39:53 INF] Using 10764 ms as testrun timeout
[16:39:56 WRN] Original project under test [path to class library I want mutated] could not be loaded.
Embedded Resources might be missing.
Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'System.Data.Entity, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
   at Mono.Cecil.BaseAssemblyResolver.Resolve(AssemblyNameReference name, ReaderParameters parameters)
   at Mono.Cecil.BaseAssemblyResolver.Resolve(AssemblyNameReference name)
   at Mono.Cecil.DefaultAssemblyResolver.Resolve(AssemblyNameReference name)
   at Mono.Cecil.MetadataResolver.Resolve(TypeReference type)
   at Mono.Cecil.ModuleDefinition.Resolve(TypeReference type)
   at Mono.Cecil.TypeReference.Resolve()
   at Mono.Cecil.Mixin.CheckedResolve(TypeReference self)
   at Mono.Cecil.SignatureReader.ReadCustomAttributeEnum(TypeReference enum_type)
   at Mono.Cecil.SignatureReader.ReadCustomAttributeElementValue(TypeReference type)
   at Mono.Cecil.SignatureReader.ReadCustomAttributeElement(TypeReference type)
   at Mono.Cecil.SignatureReader.ReadCustomAttributeFixedArgument(TypeReference type)
   at Mono.Cecil.SignatureReader.ReadCustomAttributeConstructorArguments(CustomAttribute attribute, Collection`1 parameters)
   at Mono.Cecil.MetadataReader.ReadCustomAttributeSignature(CustomAttribute attribute)
   at Mono.Cecil.CustomAttribute.<Resolve>b__35_0(CustomAttribute attribute, MetadataReader reader)
   at Mono.Cecil.ModuleDefinition.Read[TItem](TItem item, Action`2 read)
   at Mono.Cecil.CustomAttribute.Resolve()
   at Mono.Cecil.CustomAttribute.get_ConstructorArguments()
   at Mono.Cecil.ImmediateModuleReader.ReadCustomAttributes(ICustomAttributeProvider provider)
   at Mono.Cecil.ImmediateModuleReader.ReadModule(ModuleDefinition module, Boolean resolve_attributes)
   at Mono.Cecil.ImmediateModuleReader.<ReadModule>b__2_0(ModuleDefinition module, MetadataReader reader)
   at Mono.Cecil.ModuleDefinition.Read[TItem](TItem item, Action`2 read)
   at Mono.Cecil.ImmediateModuleReader.ReadModule()
   at Mono.Cecil.ModuleReader.CreateModule(Image image, ReaderParameters parameters)
   at Mono.Cecil.ModuleDefinition.ReadModule(Disposable`1 stream, String fileName, ReaderParameters parameters)
   at Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters)
   at Stryker.Core.Initialisation.EmbeddedResourcesGenerator.LoadModule(String assemblyPath, ILogger logger) in D:\a\1\s\src\Stryker.Core\Stryker.Core\Initialisation\EmbeddedResourcesGenerator.cs:line 39

**Expected behavior**
I'd like not to have these warnings and errors.

**Desktop (please complete the following information):**
Not really relevant.

**Additional context**
The error originates from Mono.Cecil, where a assembly resolver is invoked (IAssemblyResolver). The default resolver that is invoked is BaseAssemblyResolver. This resolver has some conditional compilation for .NET core vs .NET framework. 
Calling the same functionality from a .NET framework project does not result in errors. Potential solution would be to create a custom version of IAssemblyResolver that can handle both .NET core and .NET framework classes - this version must than be passed to Mono when resolving.

### PR DESCRIPTION
**Pull request by rouke-broersma**
Fix #1501 
Supersedes #1502 

@FrankvdStam I tested and cleaned up the PR. Looks good to me! I noticed you branched off of v1 but since this is a fix it can go straight to master. I created a new branch to fix this and cherry-picked your fixes onto it.

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/Initialisation/CrossPlatformAssemblyResolver.cs [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    //   Jb Evain (jbevain@gmail.com)
+    //
+    // Copyright (c) 2008 - 2015 Jb Evain
+    // Copyright (c) 2008 - 2011 Novell, Inc.
+    //
+    // Licensed under the MIT/X11 license.
+    //
+    public class CrossPlatformAssemblyResolver : IAssemblyResolver
+    {
+        private static readonly bool OnMono = Type.GetType("Mono.Runtime") != null;
```
Comment (Stijn-Rutten) [inline src/Stryker.Core/Stryker.Core/Initialisation/CrossPlatformAssemblyResolver.cs]: I'd expect this to be `_onMono`?

File: src/Stryker.Core/Stryker.Core/Initialisation/CrossPlatformAssemblyResolver.cs [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private static string GetAssemblyFile(AssemblyNameReference reference, string prefix, string gac)
+        {
+            var gac_folder = new StringBuilder()
```
Comment (Stijn-Rutten) [inline src/Stryker.Core/Stryker.Core/Initialisation/CrossPlatformAssemblyResolver.cs:382]: snake casing...

**Comment by rouke-broersma** [comment]
@FrankvdStam This fix has been released in 0.22.4. Please go ahead and test if this works for your case :)

**Comment by FrankvdStam** [comment]
I'll try but it'll be a while. Thanks anyway :)

**Pull request merged**